### DISCUSSION TITLE
How to refresh access token?

### DISCUSSION THREAD

**Discussion by khanetor** [nuxt-community/apollo-module#498] [Q&A]
We are provided with an `onLogin` helper to set a valid access token. However, that is not the only time the access token has to be set. When an access token expires, we need to use a refresh token to retrieve new access token. Is there any method to either:
1. Retrieve new token whenever a 401 network error is encountered?
2. Instead of setting the access token directly, we can specify a method `const token: () => {}` that will check the expiration time, and retrieve a new one if necessary?

**Comment by ShourovRoy** [comment]
Did you find out any solutions?

**Reply by khanetor** [reply]
> Did you find out any solutions?

I ended up not using this library at all because of this.

**Comment by RihanArfan** [comment]
This might be possible by manually implementing this with the Auth hook.  https://apollo.nuxtjs.org/recipes/authentication#auth-hook

I was wondering this today too, and from the state of issues and commits, it looks like this project is unmaintained.
